docs/book/en/src/contribute/installer.md

```diff
@@ -0,0 +1,151 @@
+# Installer Guide
+
+## Overview
+
+WasmEdge installer is designed for installing the Core Tools (`wasmedge`, `wasmedgec`), the Libraries (`libwasmedge`), the Extensions(`wasmedge-tensorflow`), and the Plugins(`wasi-nn`, `wasi-crytpo`).
+
+## Dependencies
+
+In the first version of the installer, WasmEdge provides a pure shell script implementation. However, it's not easy to maintain and is not suitable when we want to include the extensions and plugins matrix.
+
+To reduce the cost of maintenance and improve the development performance, we decided to move forward to a brand new installer that is written in Python and is compatible with both Python 2 and 3.
+
+To be compatible with the old one, we use the same entry point, `install.sh`.
+
+## Usage
+
+```bash
+curl -sSf https://raw.githubusercontent.com/WasmEdge/WasmEdge/master/utils/install.sh | bash -s -- ${OPTIONS}
+```
+
+## Roles
+
+### `install.sh`
+
+The installer entry point.
+
+#### Process
+
+1. Check if the `git` is installed; otherwise, exit with an error `Please install git`.
+2. If `PYTHON_EXECUTABLE` is given, then try to use `$PYTHON_EXECUTABLE` to execute the `install.py`. Otherwise, go to step 3.
+3. If `PYTHON_EXECUTABLE` is not set, `which` command is needed to determine the python-X executable. If it is not found installer exits else it moves on to the next step.
+4. Check if the `python3` is installed. If so, go to step 6. Otherwise, go to step 5.
+5. Check if the `python2` is installed. If so, go to step 6. Otherwise, go to step 6.
+6. Check if the `python` is installed. If so, go to step 7. Otherwise, exit with an error `Please install python or provide python path via $PYTHON_EXECUTABLE`.
+7. Print the detected python version `Using Python: $PYTHON_EXECUTABLE`.
+8. Download `install.py` with `curl` or `wget`. If the URL of `install.py` is unreachable due to a network issue, exit with an error `$INSTALL_PY_URL not reachable`. If the `curl` and `wget` are not available, exit with an error `curl or wget could not be found`.
+9. Execute the `install.py` with all received arguments.
+
+### `install.py`
+
+The real installer handles all stuff. It supports python2.7 (not tested on earlier versions) as well as the latest python versions python3.x.
+
+## Options
+
+### Help Msg
+* Short Option: `-h`
+* Full Option: `--help`
+* Description: Show this help message and exit.
+
+### Verbose
+* Short Option: `-D`
+* Full Option: `--debug`
+* Description: Enable verbosity debug
+
+### Specify Installed Version
+* Short Option: `-v VERSION`
+* Full Option: `--version VERSION`
+* Description: Install the given VERSION of WasmEdge
+* Available Value: VERSION `0.11.2` or other valid release versions.
+* Note - In case if an invalid or non existing version is suppliead the installer exits with an error.
```

Comment:
In the case of supplied an invalid or nonexistent version, the installer exists with an error.